src/handlers/messages.ts

```diff
@@ -324,6 +380,58 @@ export async function handle(rcMessage: RcMessage): Promise<void> {
     return
   }
 
+  const ts = new Date(rcMessage.ts.$date).valueOf()
+  if (rcMessage.file) {
+    if (rcMessage.attachments?.length == 1) {
+      const path = './inputs/files/' + rcMessage.file._id
+      if (!fs.existsSync(path)) {
```

Comment:
Rather than checking for an existing file, we should handle file access during the upload to prevent race conditions (unlikely, but anyhow)

Reply:
I added error handling in `uploadFile` using the `fs/promises` API in https://github.com/verdigado/rocketchat2matrix/pull/32/commits/7ce31fb48d7cad63db2e25e672322bb6efe2cdb5
I'm not completely sure if this is how it's done, so review welcome.

We should also decide on what to do if the file wasn't opened (permission, file missing): skip the message entirely or send the text message (attachment description) anyway?

Reply:
Sorry for opening [another thread](https://github.com/verdigado/rocketchat2matrix/pull/32#discussion_r1695454112), but it might be a new point.



> We should also decide on what to do if the file wasn't opened (permission, file missing): skip the message entirely or send the text message (attachment description) anyway?

I'd skip it entirely, that way it's logged and no mapping is created. So when the problem is fixed and the script re-run, it should get fixed (let's not mention the order of messages ;-))

Reply:
Ok, then nothing needs to be changed, the code returns on upload error before handling the attachment description.